var/spack/repos/builtin/packages/dd4hep/package.py

```diff
@@ -22,6 +22,10 @@ class Dd4hep(CMakePackage):
     version('1.11.0', commit='280c7d748d56a704699408ac8e57815d029b169a')
     version('1.10.0', commit='9835d1813c07d9d5850d1e68276c0171d1726801')
 
+    # Workarounds for various TBB issues in DD4hep v1.11
+    # See https://github.com/AIDASoft/DD4hep/pull/613 .
+    patch('tbb-workarounds.patch', when='@1.11.0')
```

Comment:
Shouldn't we limit this to newer TBB releases?

Reply:
I'm not sure if dd4hep will build on TBB versions before 2017.7, where the CMake files were introduced, without extra FindTBB.cmake modules from third parties. Should try to roll a build against intel-tbb@2017.6 at some point and refine tbb version requirements if need be.

Reply:
I gave it a quick try with intel-tbb@2018 and it seems to work. intel-tbb@2017 didn't work because root does not build with it and I was too lazy to try and find a working root version.

By the way, should dd4hep have an explicit dependency on intel-tbb?

Reply:
When it comes to TBB (at least), DD4hep has one of these annoying build systems that scans installed system packages and silently enables or disables features based on what it found. So the package does not strictly depend on TBB in the sense that it will build without, in sequential mode. But if it finds a TBB installation, it will use it.

We could try to add a "tbb" feature that adds a dependency on intel-tbb at the Spack level, but that would be misleading in the sense that if DD4hep finds a TBB somewhere in the dependency tree (e.g. root+tbb), it will enable the parallel features even if the "tbb" feature is disabled. We could also add an unconditional dependency on TBB, but that could be perceived as overly pessimistic.

To cleanly fix this problem would entail diving in DD4hep's build system and removing every occurence of such automagic dependency configuration in favor of explicit flags.

Reply:
I've added a dependency of dd4hep on intel-tbb on my local branch and am currently trying to run a build of `dd4hep ^ intel-tbb@2017.6 ^ root -tbb`, let's see if DD4hep can live with non-cmakified legacy versions of TBB.

Reply:
That proved to be more work than expected, but here are the results:

- As I somewhat expected dd4hep does not find TBB 2017.6, with or without this patch. It needs TBB's CMake support, which was introduced in the 2017.7 release.
- More surprisingly, 2017.7 and 2018 don't work either. Apparently, the first generation of TBB CMake config files wasn't fully baked yet. Well, the TBB release notes did warn that use of CMake was experimental at that time...
- DD4hep is not disturbed by this patch when trying to look for TBB 2017.6 or 2017.7. It silently fails either way, because that's the way its build system rolls.
- 2017.8 and 2018.1 both work with this patch, and don't work without.

So, as a tentative conclusion:

- I can disable this patch for TBB <2018.1, but there isn't really any harm in leaving it in.
- If we add a TBB dependency to DD4hep, we should probably make it `intel-tbb @2018.1:`.

Reply:
...and I checked, dd4hep builds as well with TBB 2018.1.